### DISCUSSION TITLE
Passport authenticated fail when redirect (req.user undefined)

### DISCUSSION THREAD

**Discussion by Tohru-TheTofu** [jaredhanson/passport#994] [Help]
Hi, I'm very new to web development and I'm struggling about what I do wrong here. My project have a issues when I need to find documents that have a same type with the user who log in. The first few times seem to work fine but now I'm getting a problem that req.user was undefined. When using Morgan to check, I fought that when I send the API to GET the data, isAuthenticated() turn to false and req.user was undefined. I'll show my work down, thank you so much for helping me because I did tons of research for a few days and felling so exhausted <3:

**In Index.js file, I think I called everything I need (or not)**
`
```
//Require and Configure Passport file 
const passport = require("passport")
require('./server/config/passport')(passport);

//Call flash and session
const flash = require("express-flash")
const session = require("express-session")

app.use(express.urlencoded({ extended: false }))
app.use(flash())
app.use(session({
    secret: process.env.SECRET_KEY,
    resave: false, //We won't resave the session variable if nothing changed
    saveUninitialized: false
}))
app.use(passport.initialize())
app.use(passport.session())

//Load router
app.use('/', require('./server/route/router.js'))
```

**The passport config like Stratergy, serialize or deserialize work fine in other senarios so I think I dont need to quote it here**

**And then I have a controller file to handle the method:**
```
exports.table = (req, res) => {
    console.log("Authenticated:", req.isAuthenticated());
    const userData = req.user;
    console.log(userData ,"DATABASE")
    console.log(userData.team, "TEAM")
   UserSchema.find({ "team": userData.team })
        .then(data => {
            if (!data) {
                res.status(404).send({ message: "Not found user with id " + id })
            } else {
                console.log(data)
                res.send(data)
            }
        })
        .catch(err => {
            res.status(500).send({ message: "Erro retrieving user with id " + id })
        })
};
```

**A router file to set up routes, here test is a ejs file and table is the API to get the method in the controller file above:**
```
//Routes
/**
 * @description Test Route
 * @method GET /test
 */
route.get("/test", service.testRoutes)
//--------------------------------
route.get('/table', controller.table);`
`
```
**The service.testRoutes above is actually exports from this:**
```
exports.testRoutes = (req, res) => {
    // Make a get request to /api/users
    if (!req.user) {
        res.redirect("/")
    } else {
        req.session.save(function () {
            axios.get(`http://localhost:3000/table`)
                .then(function (response) {
                    res.render('dashboard', { withCredentials: true }, { users: response.data });
                })
                .catch(err => {
                    res.send(err);
                })
        });
    }

}
```

**Comment by usualdesigner** [comment]
Check that both serializeUser and deserializeUser are wired up - a lot of people set up login and forget deserializeUser, so req.user never rehydrates on the next request even though login worked. Also check cookie settings if this is cross-origin - sameSite/secure mismatches will silently drop the session cookie on the redirect and look exactly like this.
